src/components/list-input/ListInput.tsx

```diff
@@ -1,20 +1,30 @@
-import { ChangeEvent, useState } from "react";
-import { calculateWordCount } from "../../utils";
+import {ChangeEvent, useEffect, useState} from "react";
+import {calculateWordCount, debounce} from "../../utils";
 
 interface ListInputProps {
   onAddToList: (item: string) => void;
 }
 
 export const ListInput = ({ onAddToList }: ListInputProps) => {
-  const [inputValue, setInputValue] = useState("");
+  const [inputValue, setInputValue] = useState<string>("");
+  const [currWordCount, setCurrWordCount] = useState<number>(0);
 
   function handleInputChange(event: ChangeEvent<HTMLInputElement>) {
     setInputValue(event.target.value);
   }
 
+  const debouncedCalcWordCount = debounce((text) => {
+    const wordCount = calculateWordCount(text);
+    setCurrWordCount(wordCount);
+  }, 100);
+
+  useEffect(() => {
+    debouncedCalcWordCount(inputValue);
+  }, [inputValue]);
+
```

Comment:
This is redundant. No need to complicate what is simple. I would get rid of of this debounce.

Reply:
But if there is no debounce then react would rerender each item someone types, which is unneccesarily worsens the performance. In this application, we can afford the performance loss but ive read it's best practice to use debounced when there is a component that someone will type since with debounce we are reducing the number of times react rerenders. So maybe this is overkill for this application but still it increases the performance by preventing rerenders.